### PR TITLE
use correct spectral density equivalencies for marker unit changes

### PR DESCRIPTION
**Pull request by gibsongreen**
### Description

This pull request is to address updating the spectral density equivalencies to use the original spectral axis so plugin Markers update to the correct position on the spectrum after a unit change. 

Original Behavior:

https://github.com/user-attachments/assets/e1138a05-e7f3-49aa-bb83-6877ca2969dd

Updated Behavior:

https://github.com/user-attachments/assets/00cd6394-4d81-4d66-956d-9e99f26dbe0c

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/core/marks.py [hunk 1]
```diff
@@ -127,8 +127,11 @@ def set_y_unit(self, unit=None):
 
         if self.yunit is not None and not np.all([s == 0 for s in self.y.shape]):
             if self.viewer.default_class is Spectrum1D:
+                # used to obtain spectral density equivalencies with previous data and units
+                original_spec = Spectrum1D(flux=self.y*self.yunit, spectral_axis=self.x*self.xunit)
+                eqv = u.spectral_density(original_spec.spectral_axis)
```
[the remark this answers is not in the document]
Comment (gibsongreen) [reply jdaviz/core/marks.py:132]: Will make this change right now!

**Comment by pllim** [comment]
Thanks. Is it possible to add regression test that fails on main but passes here?

**Comment by cshanahan1** [comment]
(i think this will be fixed by my related ticket btw)

**Comment by pllim** [comment]
> i think this will be fixed by my related ticket btw

Does that mean this PR is a duplicate and unnecessary?

**Comment by cshanahan1** [comment]
yes but i think its fine as a patch

**Comment by pllim** [comment]
@cshanahan1 , so is your plan to revert this in your PR, or undo your patch over there?

**Comment by cshanahan1** [comment]
> @cshanahan1 , so is your plan to revert this in your PR, or undo your patch over there?

it will just be a one line swap, and test coverage

**Comment by gibsongreen** [comment]
> Thanks. Is it possible to add regression test that fails on main but passes here?

I added some test coverage that addresess changes to the flux unit. I wrote additional coverage for translations in Cubeviz that I think should go in a small follow up after @cshanahan1  PR.

File: jdaviz/configs/default/plugins/markers/tests/test_markers_plugin.py [hunk 2]
```diff
@@ -117,6 +117,30 @@ def test_markers_cubeviz(tmp_path, cubeviz_helper, spectrum1d_cube):
...
+    uc.flux_unit.selected = 'MJy'
+
+    # set the current slice to match the markers spectral axis value
+    slice_plg = (cubeviz_helper.plugins['Slice'])
+    slice_plg.value = 4.623e-7
+
+    # get the spectrum's flux value at the current slice
+    flux_value = cubeviz_helper.get_data('Spectrum (sum)', use_display_units=True)[
+                 np.where(sv.slice_values == slice_plg.value)[0][0]
+                 ].data[0]
```
[the remark this answers is not in the document]
Comment (gibsongreen) [reply jdaviz/configs/default/plugins/markers/tests/test_markers_plugin.py]: I like this, updates are in now and its much easier to read too!

**Review by pllim — APPROVED** [review]
Thanks!

**Pull request merged**